### PR TITLE
Moving common Backendy stuff to utils in backend. Closes #852

### LINKED ISSUE #852: Merge Local and HTCondor backend common functionality
Merge Local and HTCondor backend common functionality.
Move also WorkflowPaths to Backend component.

### PR DESCRIPTION
**Pull request by gauravs90**
Moves some of the common stuff shared across backends using the shared file system (e.g. LocalBackend and HTCondorBackend) in the utils package in the backend project.

### CODE REVIEW

**Comment by gauravs90** [comment]
I have not gone deep into the individual BackendJobExecutionActors.. I can do it if reviewers think so, but honestly I think that's an overkill right now.

**Comment by gauravs90** [comment]
@geoffjentry and @francares

File: backend/src/main/scala/cromwell/backend/utils/SharedFsExpressionFunctions.scala [hunk 1]
```diff
@@ -11,35 +11,40 @@ import wdl4s.values.{WdlFile, WdlValue}
 import scala.language.postfixOps
 import scala.util.{Success, Try}
 
-object LocalJobExpressionFunctions {
+object SharedFsExpressionFunctions {
```
Comment (francares) [inline backend/src/main/scala/cromwell/backend/utils/SharedFsExpressionFunctions.scala]: I think this should be placed in an "io" package.

File: backend/src/main/scala/cromwell/backend/utils/WorkflowPaths.scala [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package cromwell.backend.impl.local
+package cromwell.backend.utils
```
Comment (francares) [inline backend/src/main/scala/cromwell/backend/utils/WorkflowPaths.scala]: Same here.

**Comment by francares** [comment]
:+1:

**Comment by geoffjentry** [comment]
I think this is fine but I'd ask that you et another reviewer, I'm without wifi for most of the day and iPhones aren't the best code review tool :)

**Comment by gauravs90** [comment]
@geoffjentry Gotcha! I'll nominate @Horneth then since he's been familiar with this part of the code.

File: backend/src/main/scala/cromwell/backend/io/SharedFsExpressionFunctions.scala [hunk 3]
```diff
@@ -11,35 +11,40 @@ import wdl4s.values.{WdlFile, WdlValue}
...
       jobPaths.stdout.toAbsolutePath.toString,
       jobPaths.stderr.toAbsolutePath.toString
     )
-
-    new LocalJobExpressionFunctions(LocalJobExecutionActor.fileSystems, callContext)
+    apply(callContext, fileSystems)
   }
+
+  def apply(callContext: CallContext, fileSystems: List[FileSystem]): SharedFsExpressionFunctions =
+    new SharedFsExpressionFunctions(fileSystems, callContext)
```
Comment (Horneth) [inline backend/src/main/scala/cromwell/backend/io/SharedFsExpressionFunctions.scala]: Do we need this apply ?

File: backend/src/main/scala/cromwell/backend/io/WorkflowPaths.scala [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package cromwell.backend.impl.local
+package cromwell.backend.io
```
Comment (Horneth) [inline backend/src/main/scala/cromwell/backend/io/WorkflowPaths.scala]: I think we could break out `JobPaths` into its own file. It's confusing to have both WorkflowPaths and JobPaths in `WorkflowPaths.scala`

**Comment by Horneth** [comment]
👍

File: backend/src/main/scala/cromwell/backend/io/SharedFsExpressionFunctions.scala (hunk 3 above)
Comment (gauravs90) [inline backend/src/main/scala/cromwell/backend/io/SharedFsExpressionFunctions.scala]: Nope, we don't. Removed it.

File: backend/src/main/scala/cromwell/backend/io/WorkflowPaths.scala (hunk 4 above)
Comment (gauravs90) [inline backend/src/main/scala/cromwell/backend/io/WorkflowPaths.scala]: Done.

**Pull request merged**